src/test/java/net/dancier/kikeriki/state/DancerStateMachineTest.java

```diff
@@ -48,9 +41,11 @@ public void recentUnreadMessageLeadsToScheduledRecheck() {
     dancerStateChatMessagesRepository.setLastUnreadMessageAt(dancerId, lastUnreadMessage);
     dancerStateMailMessageRepository.setLastMailSendAt(dancerId, lastMailMessageSend);
 
-    dancerStateMachine.check(dancerId);
+    Optional<SendMail> result = dancerStateMachine.check(dancerId);
 
-    Mockito.verify(applicationEventPublisher, times(1)).publishEvent(new SendMail());
+    assertThat(result.isPresent())
+      .describedAs("A SendMail event should be present")
+      .isTrue();
```

Comment:
First you have to fight your way through all reviews including the last one.
As said before: a discussion on the changes of the older PRs would be a waste of time, since they do not represent the current state of the implementation. They are only still there, to help you, undertand the past development steps.

Considering your question:
How the dancer is involved in the end, is _not_ implemented yet. We can discuss this on the current version.
And:
In my humble opinion, one would not gain much (any?) benefit from decoupling the state machine from the actual involvement by sending an application-event. It just hides what is realy happening. I cannot see the benefit here. The decoupling, that counts at that point is, that the actual involvement (aka Sending the mail etc.) does not know the state machine. That way, it can be testet separatly. A simple interface, that can be mocked during tests of the state machine, would be enough decoupling for any purpose, i am able to imagine here...